spynnaker/pyNN/models/neuron/synaptic_matrices.py

```diff
@@ -152,50 +153,56 @@ def __app_matrix(self, app_edge, synapse_info):
         self.__matrices[key] = matrix
         return matrix
 
-    def synapses_size(self, app_edges):
+    def synapses_size(self, edges):
         """ The size of the synaptic blocks in bytes
 
-        :param iterable(~pacman.model.graphs.application.ApplicationEdge) \
-                app_edges:
-            The incoming application edges
+        :param edges: The incoming edges
+        :type edges: iterable(~pacman.model.graphs.AbstractEdge)
         :rtype: int
         """
         # Base size requirements
         # 1 word for address of direct addresses, and
         # 1 word for the size of the direct addresses matrix in bytes
         memory_size = 2 * BYTES_PER_WORD
-        for in_edge in app_edges:
-            if isinstance(in_edge, ProjectionApplicationEdge):
-                for synapse_info in in_edge.synapse_information:
-                    matrix = self.__app_matrix(in_edge, synapse_info)
+        for in_edge in edges:
+            if isinstance(in_edge, MachineEdge):
+                app_edge = in_edge.app_edge
+            else:
+                app_edge = in_edge
+            if isinstance(app_edge, ProjectionApplicationEdge):
+                for synapse_info in app_edge.synapse_information:
+                    matrix = self.__app_matrix(app_edge, synapse_info)
                     memory_size = matrix.add_matrix_size(memory_size)
                     memory_size = matrix.add_delayed_matrix_size(memory_size)
         return memory_size
 
-    def size(self, app_edges):
+    def size(self, edges):
         """ The size required by all parts of the matrices
 
-        :param iterable(~pacman.model.graphs.application.ApplicationEdge) \
-                app_edges:
-            The incoming application edges
+        :param iterable(~pacman.model.graphs.AbstractEdge) edges:
+            The incoming edges
         :rtype: int
         """
         return (
-            self.synapses_size(app_edges) +
-            self.__gen_info_size(app_edges) +
-            self.__poptable.get_master_population_table_size(app_edges))
+            self.synapses_size(edges) +
+            self.__gen_info_size(edges) +
+            self.__poptable.get_master_population_table_size(edges))
 
-    def __gen_info_size(self, app_edges):
+    def __gen_info_size(self, edges):
         """ The size in bytes of the synaptic expander parameters
 
-        :param iterable(~pacman.model.graphs.application.ApplicationEdge) \
-                app_edges:
-            The incoming application edges
+        :param edges: The incoming edges
+        :type edges: iterable(~pacman.model.graphs.AbstractEdge)
         :rtype: int
         """
         gen_on_machine = False
         size = 0
-        for app_edge in app_edges:
+        for edge in edges:
+            if isinstance(edge, MachineEdge):
+                app_edge = edge.app_edge
+            else:
+                app_edge = edge
+
```

Comment:
Like the above, I can see this going wrong...

Reply:
i know!